### PR TITLE
Refactored ratings to use svg in data uris #273, dev-36979

### PR DESCRIPTION
**Pull request by Januszpl**
Refs: #273 
internal: dev-36979
I used icons from design (svg, optimize them and encode to use in data uri) more comments in README.md

### CODE REVIEW

File: components/02-elements/rating/_rating.scss [hunk 1]
```diff
@@ -1,78 +1,105 @@
-$rating-font-size: $font-size-large;
-$rating-active   : $orange;
+$rating__size: 96px;
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: missing `!deafult` flag

File: components/02-elements/rating/_rating.scss [hunk 2]
```diff
@@ -1,78 +1,105 @@
...
-    span {
-        display: inline-block;
+    &--rate {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        direction: rtl;
+    }
+    &__rate-item {
+        flex: 1 0 ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: spaces around operator, IMO should be like : `flex: 1 0 ($rating__size / 5);`

File: components/02-elements/rating/_rating.scss [hunk 3]
```diff
@@ -1,78 +1,105 @@
...
-
-        &:before {
-            content: '\2605' '\2605' '\2605' '\2605' '\2605';
-            position: absolute;
-            left: 0;
-            width: 100%;
-            height: 100%;
+        width: $rating__size;
+        &--rate {
+            width: ($rating__size/5)
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: soace around operator and semi-colon on the end missing

File: components/02-elements/rating/_rating.scss [hunk 4]
```diff
@@ -1,78 +1,105 @@
...
+        }
+        &:before { // inactive
+            background-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 24 19' xmlns='http://www.w3.org/2000/svg' fill-rule='evenodd' clip-rule='evenodd' stroke-linejoin='round' stroke-miterlimit='1.414'%3E%3Cpath fill='none' d='M-2-2h24v24H-2z'/%3E%3Cpath d='M20 7.24l-7.19-.62L10 0 7.19 6.63 0 7.24l5.46 4.73L3.82 19 10 15.27 16.18 19l-1.63-7.03L20 7.24zM10 13.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L10 4.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L10 13.4z' fill='%23c2c1df'/%3E%3C/svg%3E ");
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: we can add files like that inside css that shouldn't be put in the separate sprite readable for css background? I'm not saying it's a bug, just curious :)

File: components/02-elements/rating/_rating.scss [hunk 5]
```diff
@@ -1,78 +1,105 @@
...
+            background-repeat: repeat-x;
+            background-position: left center;
+            background-size: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator... maybe put it in variable?

File: components/02-elements/rating/_rating.scss [hunk 6]
```diff
@@ -1,78 +1,105 @@
...
+            bottom: 0;
+            content: "";
             display: block;
-            z-index: 1;
-            color: $gray;
+            height: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 7]
```diff
@@ -1,78 +1,105 @@
...
+            position: absolute;
+            right: 0;
+            top: 0;
+            width: $rating__size;
         }
-    }
 
-    &__fill {
+        &--single {
+            width: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 8]
```diff
@@ -1,78 +1,105 @@
...
+        }
+    }
+    &__indicator {
         display: block;
+        text-indent: -10000px;
+        width: $rating__size;
+        height: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 9]
```diff
@@ -1,78 +1,105 @@
...
-        &:before {
-            content: '\2605' '\2605' '\2605' '\2605' '\2605';
-            position: relative;
+        left: 0;
+        top: 0;
+        &:before { // active
+            background-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 24 19' xmlns='http://www.w3.org/2000/svg' fill-rule='evenodd' clip-rule='evenodd' stroke-linejoin='round' stroke-miterlimit='1.414'%3E%3Cpath fill='none' d='M-2-2h24v24H-2z'/%3E%3Cpath d='M10 15.27L16.18 19l-1.64-7.03L20 7.24l-7.19-.61L10 0 7.19 6.63 0 7.24l5.46 4.73L3.82 19 10 15.27z' fill='%23fab216'/%3E%3C/svg%3E ");
+            background-repeat: repeat-x;
+            background-position: left center;
+            background-size: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 10]
```diff
@@ -1,78 +1,105 @@
...
+            bottom: 0;
+            content: "";
+            display: block;
+            height: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 11]
```diff
@@ -1,78 +1,105 @@
...
+            right: 0;
+            text-indent: 10000px;
             top: 0;
-            width: 100%;
-            height: 100%;
-            display: block;
-            z-index: 2;
-            color: $rating-active;
-        }
+            z-index:3;
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: missing space after colon

File: components/02-elements/rating/_rating.scss [hunk 12]
```diff
@@ -1,78 +1,105 @@
...
 
-        & * {
+            .rating__star--rate:hover & {
+                display: block;
+            }
+        }
+        &:after { //active backgroun
+            background-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 24 19' xmlns='http://www.w3.org/2000/svg' fill-rule='evenodd' clip-rule='evenodd' stroke-linejoin='round' stroke-miterlimit='1.414'%3E%3Cpath fill='none' d='M-2-2h24v24H-2z'/%3E%3Cpath d='M20 7.24l-7.19-.62L10 0 7.19 6.63 0 7.24l5.46 4.73L3.82 19 10 15.27 16.18 19l-1.63-7.03L20 7.24zM10 13.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L10 4.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L10 13.4z' fill='%23fab216'/%3E%3C/svg%3E ");
+            background-repeat: repeat-x;
+            background-size: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 13]
```diff
@@ -1,78 +1,105 @@
...
+            background-position: left center;
+            bottom: 0;
+            content: "";
+            display: block;
+            height: ($rating__size/5);
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: space around operator.

File: components/02-elements/rating/_rating.scss [hunk 14]
```diff
@@ -1,78 +1,105 @@
...
+            left: 0;
             position: absolute;
-            opacity: 0;
+            right: 0;
+            text-indent: 10000px;
             top: 0;
-            width: 0;
-            height: 0;
+            z-index:2;
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: missing space after colon

File: components/02-elements/rating/rating--summary.hbs [hunk 15]
```diff
@@ -1,5 +1,5 @@
-<div class="rating__summary {{ class }}" {{{ attributes }}}>
-    <span class="rating__fill" {{{ percentFilled }}}>
-        {{{ percent }}}
-    </span>
+<div class="rating rating--summary {{ class }}" {{{ attributes }}}>
+    <div class="rating__star {{ star.class }}" {{{ star.attributes }}}>
+        <span class="rating__indicator {{ indicator.class }}" {{{ indicator.attributes }}}></span>
+    </div>
```
Comment (anqaka) [inline components/02-elements/rating/rating--summary.hbs]: Recently we were talking about rules order inside styles - if you have some time please check the order of the rules and change it where possible :) (not a bug)

**Comment by anqaka** [comment]
update changelog ;)

File: components/02-elements/rating/rating--summary.hbs (hunk 15 above)
Comment (Januszpl) [reply components/02-elements/rating/rating--summary.hbs]: changed - but I think aZ with automation would be better solution or by group (but make it do by machine not us :D)

File: components/02-elements/rating/_rating.scss (hunk 14 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: thx. fixed

File: components/02-elements/rating/_rating.scss (hunk 13 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: changed to variable

File: components/02-elements/rating/_rating.scss (hunk 12 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: changed to variable

File: components/02-elements/rating/_rating.scss (hunk 11 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: changed to variable

Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: fixed

File: components/02-elements/rating/_rating.scss (hunk 4 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: For this use case I think it's easier for update and maintance just to replace svg here as it's in readme. However if you or anyone can create working solution with your proposition then we could add it of course

File: components/02-elements/rating/_rating.scss [hunk 16]
```diff
@@ -1,78 +1,103 @@
-$rating-font-size: $font-size-large;
-$rating-active   : $orange;
-
+$rating__size: 96px;
+$rating__item-size: $rating__size / 5;
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: missing `!deafult` flag

File: components/02-elements/rating/_rating.scss [hunk 17]
```diff
@@ -1,78 +1,103 @@
...
-            height: 100%;
+            top: 0;
+            right: 0;
             display: block;
-            z-index: 1;
-            color: $gray;
+            width: $rating__size;
+            content: "";
+            height: $rating__item-size;
+            background-image: url("raw/components/01-globals/icons/files/star-border.svg");
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: Icon not found in preview... IMO you have to add separate icon to `public/images`but I'm wondering if we can use here icon inline? I recommend build short sprite with icons used as css background

File: components/02-elements/rating/_rating.scss (hunk 1 above)
Comment (anqaka) [reply components/02-elements/rating/_rating.scss]: still missing :)

File: components/02-elements/rating/_rating.scss [hunk 18]
```diff
@@ -1,78 +1,108 @@
-$rating-font-size: $font-size-large;
-$rating-active   : $orange;
+$rating__size: 96px;
+$rating__item-size: $rating__size / 5;
+$icon__active-color: '#fab216';
+$icon__active-color--with-border: '#fab216';
+$icon__inactive-color: '#c2c1df';
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: missing `!default` flag

File: components/02-elements/rating/_rating.scss [hunk 19]
```diff
@@ -1,78 +1,108 @@
...
     cursor: pointer;
-    text-align: left;
 
-    span {
-        display: inline-block;
+    &--rate {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        direction: rtl;
```
Comment (anqaka) [inline components/02-elements/rating/_rating.scss]: you have two modifiers on main element: `--rate` and `--summary`, IMO, you should delete one and style it for default... I cannot see the `--summary` so maybe should be deleted from hbs template? But IMO, it's not quite logic that you have `summary` variant and in css `--rate` (and not `--summary` modifier). And do you think?

File: components/02-elements/rating/rating--summary.hbs (hunk 15 above)
Comment (anqaka) [inline components/02-elements/rating/rating--summary.hbs]: I'm wondering - it looks like we have `--rate` variant and modifier and not `--summary` Maybe will be better to refactor it to be more readable?

File: docs/styles/_components.scss [hunk 20]
```diff
@@ -6,6 +6,7 @@
 @import '../../components/01-globals/mixins/clearfix';
 @import '../../components/01-globals/mixins/grid-column-width';
 @import '../../components/01-globals/mixins/mq';
+@import '../../components/01-globals/mixins/svg-uri';
```
Comment (anqaka) [inline docs/styles/_components.scss]: I'm not convenience about this solution, but I'm not saying it's bad. I think with sprote for css background will be better b/c using a lot of svg inside the css we can increase amount of code and time of generate. But we can try this attitude :)

Comment (Januszpl) [reply docs/styles/_components.scss]: as I said we can have control of color here. If you would like to show sprite solution for rating feel free as my try with svg for this case didn't convinced me (but also maybe I make sth wrong)

File: components/02-elements/rating/_rating.scss (hunk 19 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: yes. after refactoring changing variants is good idea. done

File: components/02-elements/rating/_rating.scss (hunk 18 above)
Comment (Januszpl) [reply components/02-elements/rating/_rating.scss]: done plus added aligment

File: docs/styles/_components.scss (hunk 20 above)
Comment (anqaka) [reply docs/styles/_components.scss]: I think we should try with sprite but we can do it later, I've create a separate issue: #277

**Pull request merged**